gst/nnstreamer/elements/gsttensor_debug.c

```diff
@@ -0,0 +1,441 @@
+{
+  static GType type = G_TYPE_INVALID;
+
+  if (type == G_TYPE_INVALID) {
+    static const GFlagsValue values[] = {
+      {C_FLAGS (TDBG_META_DISABLED),
+          "Do not log stream metadata.", "disabled"},
+      {C_FLAGS (TDBG_META_TIMESTAMP), "Log timestamp information", "timestamp"},
+      {C_FLAGS (TDBG_META_QUERYSERVER),
+          "Log tensor-query-server related information", "queryserver"},
+      {0, NULL, NULL}
+    };
+    type = g_flags_register_static ("gtd_meta", values);
+  }
+  return type;
+}
+
+#define DEFAULT_TENSOR_DEBUG_META_FLAGS (TDBG_META_DISABLED)
+
+/**
+ * @brief Flag to print minimized log.
+ */
+#define DEFAULT_SILENT TRUE
+
+#define gst_tensor_debug_parent_class parent_class
+G_DEFINE_TYPE (GstTensorDebug, gst_tensor_debug, GST_TYPE_BASE_TRANSFORM);
+
+/* gobject vmethods */
+static void gst_tensor_debug_set_property (GObject * object,
+    guint prop_id, const GValue * value, GParamSpec * pspec);
+static void gst_tensor_debug_get_property (GObject * object,
+    guint prop_id, GValue * value, GParamSpec * pspec);
+static void gst_tensor_debug_finalize (GObject * object);
+
+/* gstbasetransform vmethods */
+static GstFlowReturn gst_tensor_debug_transform_ip (GstBaseTransform * trans,
+    GstBuffer * buffer);
+static GstCaps *gst_tensor_debug_fixate_caps (GstBaseTransform * trans,
+    GstPadDirection direction, GstCaps * caps, GstCaps * othercaps);
+static gboolean gst_tensor_debug_set_caps (GstBaseTransform * trans,
+    GstCaps * incaps, GstCaps * outcaps);
+
+/**
+ * @brief Initialize the tensor_debug's class.
+ */
+static void
+gst_tensor_debug_class_init (GstTensorDebugClass * klass)
+{
+  GObjectClass *object_class;
+  GstElementClass *element_class;
+  GstBaseTransformClass *trans_class;
+
+  GST_DEBUG_CATEGORY_INIT (gst_tensor_debug_debug, "tensor_debug", 0,
+      "Element to provide debug information of other/tensors streams. If this is enabled, the pipeline performance and overhead may be deteriorated significantly.");
+
+  trans_class = (GstBaseTransformClass *) klass;
+  object_class = (GObjectClass *) klass;
+  element_class = (GstElementClass *) klass;
+
+  /* GObjectClass vmethods */
+  object_class->set_property = gst_tensor_debug_set_property;
+  object_class->get_property = gst_tensor_debug_get_property;
+  object_class->finalize = gst_tensor_debug_finalize;
+
+  /**
+   * GstTensorDebug::silent:
+   *
+   * The flag to enable/disable debugging messages.
+   */
+  g_object_class_install_property (object_class, PROP_SILENT,
+      g_param_spec_boolean ("silent", "silent", "Produce verbose output",
+          DEFAULT_SILENT, G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
+
+  /**
+   * GstTensorDebug::output:
+   *
+   * The combination of enums configuring output methods.
+   * @todo check the behavior of name and nick (output methods vs output)
+   */
+  g_object_class_install_property (object_class, PROP_OUTPUT,
+      g_param_spec_flags ("output methods", "output",
```

Comment:
The name should not contain spaces. May use '-' or '_' for this name and others.